core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -90,11 +89,10 @@ public KafkaStore(SchemaRegistryConfig config,
     this.topic = config.getString(SchemaRegistryConfig.KAFKASTORE_TOPIC_CONFIG);
     this.desiredReplicationFactor =
         config.getInt(SchemaRegistryConfig.KAFKASTORE_TOPIC_REPLICATION_FACTOR_CONFIG);
-    int port = KafkaSchemaRegistry.getSchemeAndPortForIdentity(
-        config.getInt(SchemaRegistryConfig.PORT_CONFIG),
-        config.getList(RestConfig.LISTENERS_CONFIG),
-        config.interInstanceProtocol()
-    ).port;
+    this.config = config;
+    int port = KafkaSchemaRegistry.getInternalListener(config.getListeners(),
+        config.interInstanceListenerName(),
+        config.interInstanceProtocol()).getUri().getPort();
     this.groupId = config.getString(SchemaRegistryConfig.KAFKASTORE_GROUP_ID_CONFIG).isEmpty()
                    ? String.format("schema-registry-%s-%d",
                         config.getString(SchemaRegistryConfig.HOST_NAME_CONFIG), port)
```

Comment:
Should the host be taken from
```
String host =   internalListenerName.equals(interInstanceListenerNameConfig)
                     ? internalListener.getUri().getHost()
                     : config.getString(SchemaRegistryConfig.HOST_NAME_CONFIG);
```

Reply:
This is the consumer groupId that is used to read the _schemas topic and independent from myidentity that is used to identify an instance for leader election. Pasting the relevant code for identity 

Setting my identity:
https://github.com/confluentinc/schema-registry/blob/cf960b385b2f2e5cd186e6468029ecc3eefd0bc8/core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java#L164

Registering identity with the group coordinator:
[`leaderElector = new KafkaGroupLeaderElector(config, myIdentity, this);`](https://github.com/confluentinc/schema-registry/blob/cf960b385b2f2e5cd186e6468029ecc3eefd0bc8/core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java#L331)

Leader election result:
https://github.com/confluentinc/schema-registry/blob/cf960b385b2f2e5cd186e6468029ecc3eefd0bc8/core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java#L379

Reply:
I did not want to change the existing groupId logic due to changes in identity due to named listeners - IIUC, they need not match. WDYT?
https://github.com/confluentinc/schema-registry/blob/cf960b385b2f2e5cd186e6468029ecc3eefd0bc8/core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java#L98

Reply:
The logic for groupId can be different then identity, but one concern is that HOST_NAME_CONFIG will no longer be mandatory in which case all 3 instances of SR will get same identity and their kafka consumers will not work as intended.

Reply:
HOST_NAME_CONFIG defaults to the system host name `InetAddress.getLocalHost().getCanonicalHostName()` when not explicitly provided - so this will continue to be unique (as long as brokers are configured with different names, which is going to be the case). I prefer to keep this logic unchanged for now to avoid any surprises, since it really does not depend on the listener changes in this PR.

Reply:
ohh. I remember you mentioning this yesterday, but totally forgot while reviewing this PR again :(